pkg/netstore/netstore_test.go

```diff
@@ -187,15 +245,36 @@ func (r *retrievalMock) RetrieveChunk(ctx context.Context, addr swarm.Address) (
 }
 
 type mockRecovery struct {
-	hookC chan bool
+	hookC     chan bool
+	sendChunk swarm.Chunk
 }
 
 // Send mocks the pss Send function
-func (mr *mockRecovery) recovery(chunkAddress swarm.Address, targets trojan.Targets) error {
+func (mr *mockRecovery) recovery(chunkAddress swarm.Address, targets trojan.Targets, chunkC chan swarm.Chunk) (swarm.Address, error) {
 	mr.hookC <- true
-	return nil
+	if mr.sendChunk != nil {
+		sch := getSocChunk(mr.sendChunk)
+		chunkC <- sch
+		return sch.Address(), nil
+	}
+	return swarm.NewAddress([]byte("deadbeef")), nil
 }
 
 func (r *mockRecovery) RetrieveChunk(ctx context.Context, addr swarm.Address) (chunk swarm.Chunk, err error) {
 	return nil, fmt.Errorf("chunk not found")
 }
+
+func getSocChunk(c swarm.Chunk) swarm.Chunk {
+	var dummyChunk swarm.Chunk
+	privKey, err := crypto.GenerateSecp256k1Key()
+	if err != nil {
+		return dummyChunk
+	}
+	signer := crypto.NewDefaultSigner(privKey)
+	id := make([]byte, 32)
+	sch, err := soc.NewChunk(id, c, signer)
+	if err != nil {
+		return dummyChunk
+	}
+	return sch
+}
```

Comment:
It would be nice that the errors are propagated, or at least to have this function as test helper with errors handled.